source/extensions/filters/network/redis_proxy/hotkey/cache/cache.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+
+#include "absl/container/flat_hash_map.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace RedisProxy {
+namespace HotKey {
+namespace Cache {
+namespace LFUCache {
+class LFUCache;
+} // namespace LFUCache
+
+class Cache {
+  friend class LFUCache::LFUCache;
+
+public:
+  Cache(const uint8_t& capacity, const uint8_t& warming_capacity = 5)
```

Comment:
perf nit:

```suggestion
  Cache(uint8_t capacity, uint8_t warming_capacity = 5)
```

Below also you are passing some small values by const ref. In general it's faster to pass these by value so I would clean that up as well.